File: 프로그래머스/lv1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    // 양쪽 순서 학생에게만 체육복을 빌려줄 수 있음
    // 전체 학생 수 n, 체육복 도난 학생 번호 lost, 여벌의 체육복 가져온 학생들 번호 reserve
    let students = [];
    for (let i = 1; i <= n; i++) {
        students.push(i);
    }
    
    // 도난 당한 학생 제외
    const haveClothes = students.filter((x => !lost.includes(x)));
    let count = haveClothes.length;
    
    // 체육복 빌려주기
    console.log(haveClothes);
    for (let j = 0; j < count; j++) {
        if (lost.includes(haveClothes[j] - 1)) {
            count++;
            lost.splice(j-1, 1)
        } else if (lost.includes(haveClothes[j] + 1)) {
            count++;
            lost.splice(j+1, 1)
        }
        // else {
        //     return count;
        // }
    }
    return count;
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    students = [i for i in range(1, n + 1)]

    students = list(set(students) - set(lost))

    for cloth in reserve:
        if cloth not in students:
            students.append(cloth)

    reserve = list(set(reserve) - set(lost))

    for cloth in reserve:
        if (cloth - 1) >= 1 and (cloth - 1) not in students:
            students.append(cloth - 1)
        elif (cloth + 1) <= n and (cloth + 1) not in students:
            students.append(cloth + 1)

    return len(students)